atest/robot/output/listener_interface/support_variables_in_test_case_name_with_listener_v2.robot

```diff
@@ -0,0 +1,32 @@
+*** Settings ***
+Suite Setup      Run Tests    --listener ${DATADIR}/output/listeners/verify_var_name_in_test_case_v2.py    output/listeners/support_variables_in_test_case_name.robot
+Resource         atest_resource.robot
+
+
+*** Variables ***
+${MESSAGE FILE}    %{TEMPDIR}${/}messages.txt
+
+*** Test Cases ***
+Test Case Name With Variable
+    Check Test Case    ${TESTNAME}
+    Check Stdout Contains    SEPARATOR=\n
+    ...    [Start] [name] ${TESTNAME} [original name] Test Case \${NAME} With \${VARIABLE}
+    ...    [END] [name] ${TESTNAME} [original name] Test Case \${NAME} With \${VARIABLE}
+
+Test Case Name With ['My', 'List']
```

Comment:
Test names on the `atest/robot` side should be easy to understand. These tests are great otherwise and I can change these myself after merging the PR.